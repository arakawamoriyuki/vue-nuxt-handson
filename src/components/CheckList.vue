<template>
  <v-list subheader two-line flat>
    <v-subheader>{{ title }}</v-subheader>
    <v-list-item-group multiple>
      <v-list-item v-for="item in items" :key="item.id">
        <v-list-item-action>
          <v-checkbox
            v-model="item.done"
            color="primary"
            @click="onCheck(item)"
          ></v-checkbox>
        </v-list-item-action>
        <v-list-item-content>
          <v-list-item-title>{{item.id}}. {{item.title}}</v-list-item-title>
          <v-list-item-subtitle>{{item.description}}</v-list-item-subtitle>
        </v-list-item-content>
        <v-list-item-action>
          <v-icon color="red" @click="onDelete(item)">mdi-delete</v-icon>
        </v-list-item-action>
      </v-list-item>
    </v-list-item-group>
  </v-list>
</template>

<script>
export default {
  name: 'check-list',
  props: {
    title: String,
    items: Array,
    onCheck: Function,
    onDelete: Function,
  },
};
</script>
