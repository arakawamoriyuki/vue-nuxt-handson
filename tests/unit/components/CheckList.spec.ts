import { shallowMount } from '@vue/test-utils';
import CheckList from '@/components/CheckList.vue';

describe('CheckList', () => {
  it('CheckListが表示される', () => {
    const wrapper = shallowMount(CheckList);
    expect(wrapper).toMatchSnapshot();
  });
});
