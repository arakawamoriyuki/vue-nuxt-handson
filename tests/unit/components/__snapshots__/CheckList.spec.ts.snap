// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckList CheckListが表示される 1`] = `
<v-list-stub tag="div" flat="true" subheader="true" twoline="true">
  <v-subheader-stub></v-subheader-stub>
  <v-list-item-group-stub activeclass="v-item--active" multiple="true"></v-list-item-group-stub>
</v-list-stub>
`;
